### DISCUSSION TITLE
How to fix Startup Session breaking ability to run commands in new Kitty session?

### DISCUSSION THREAD

**Discussion by NovaViper** [kovidgoyal/kitty#5474] [Q&A] [answered]
Hey guys! I'm new to using Kitty and I'm wondering what I'm doing wrong. So I made a startup session that creates 3 kitty windows within the same tab in the tall layout. Two of the kitty windows opening up the zsh prompt and the third one opening cmatrix to run some decorative Matrix rain. But since doing this, I've noticed if I have a program that launches a terminal command by creating a new terminal window, the commands do not get ran but instead they run my startup script.. is there anyway to have kitty pass on the commands that programs send to the terminal? My config is the default configuration with no other options enabled other than the `startup_session` config. My system's running EndeavourOS with Kitty 0.26.1

I've done an example with `kitty cmatrix` that should open another Kitty window and then run cmatrix (producing the matrix screensaver); this works when I don't have `startup_session` enabled but stops working when I have that config enabled (all it does it open my startup session but does not run the command).

This is what my startup session file (called startup.conf) looks like!
```
# Set the layout for the current tab
layout tall
enabled_layouts tall:bias=55;full_size=1;mirrored=false
# Create a window and run the specified command in it
launch
# Create a window with some environment variables set and run vim in it
launch
# Set the title for the next window
launch cmatrix -ba -u 3 -C blue
```

**Comment by kovidgoyal** [comment]
If you want your session to work with just one kitty window, run that
one kitty window using --session dont put it in kitty.conf.

On Sun, Sep 04, 2022 at 08:55:25PM -0700, Nova Leary wrote:
> Hey guys! I'm new to using Kitty and I'm wondering what I'm doing wrong. So I made a startup session that creates 3 kitty windows within the same tab in the tall layout. Two of the kitty windows opening up the zsh prompt and the third one opening cmatrix to run some decorative Matrix rain. But since doing this, I've noticed if I have a program that launches a terminal command by creating a new terminal window, the commands do not get ran but instead they run my startup script.. is there anyway to have kitty pass on the commands that programs send to the terminal? My config is the default configuration with no other options enabled other than the `startup_session` config. My system's running EndeavourOS with Kitty 0.26.1
> 
> I've done an example with `kitty cmatrix' that should open another Kitty window and then run cmatrix (producing the matrix screensaver); this works when I don't have `startup_session` enabled but stops working when I have that config enabled (all it does it open my startup session but does not run the command).
> 
> This is what my startup session file (called startup.conf) looks like!
> ```
> # Set the layout for the current tab
> layout tall
> enabled_layouts tall:bias=55;full_size=1;mirrored=false
> # Create a window and run the specified command in it
> launch
> # Create a window with some environment variables set and run vim in it
> launch
> # Set the title for the next window
> launch cmatrix -ba -u 3 -C blue
> ```
> 
> -- 
> Reply to this email directly or view it on GitHub:
> https://github.com/kovidgoyal/kitty/discussions/5474
> You are receiving this because you are subscribed to this thread.
> 
> Message ID: ***@***.***>

-- 
_____________________________________

Dr. Kovid Goyal 
https://www.kovidgoyal.net
https://calibre-ebook.com
_____________________________________

**Reply by NovaViper** [reply]
Ah ok! Is there a way to have my session open up when I open the task icon for it? I want to be able to open the session when I select the task icon but not open the session when other programs call kitty.

**Comment by kovidgoyal** [comment, marked as answer]
create a launcher to run kitty with whatever command line arguments you
want, or alternately, use https://sw.kovidgoyal.net/kitty/faq/#how-do-i-specify-command-line-options-for-kitty-on-macos

**Reply by NovaViper** [reply]
Ah, I got it working with a desktop shortcut as suggested, and even went and added it to my Ansible script so when I setup my system it automatically puts my launcher in its correct place, Thanks!
